src/lib/installPlugin.js

```diff
@@ -139,3 +155,99 @@ async function createFileRecursive(parent, dir) {
 		await createFileRecursive(newParent, dir);
 	}
 }
+
+/** Resolve dependency
+ * @param {string} id
+ * @param {(message: any) => void} setMessage
+ * @returns {Promise<boolean>} has error
+ */
+async function resolveDependency(id, setMessage) {
+	let purchaseToken;
+	let product;
+	let isPaid = false;
+
+	try {
+		const remoteDependency = await fsOperation(
+			constants.API_BASE,
+			`plugin/${id}`,
+		)
+			.readFile("json")
+			.catch(() => null);
+
+		if (!remoteDependency) return true;
+
+		const version = await isInstalled(id);
+		if (remoteDependency?.version === version) return false;
+
+		plugin = Object.assign({}, remoteDependency);
```

Comment:
i guess you forget to define this variable

Reply:
The variable is unused